Extension/src/LanguageServer/extension.ts

```diff
@@ -1162,3 +1163,36 @@ export function UpdateInsidersAccess(): void {
         void vscode.commands.executeCommand("workbench.extensions.installExtension", "ms-vscode.cpptools", { installPreReleaseVersion: true }).then(undefined, logAndReturn.undefined);
     }
 }
+
+export async function preReleaseCheck(): Promise<void> {
+    const displayedPreReleasePrompt: PersistentState<boolean> = new PersistentState<boolean>("CPP.displayedPreReleasePrompt", false);
+
+    // First we need to make sure the user isn't already on a pre-release version and hasn't dismissed this prompt before.
+    if (!displayedPreReleasePrompt.Value && util.getCppToolsTargetPopulation() === TargetPopulation.Public) {
+        // Get the info on the latest version from the marketplace to check if there is a pre-release version available.
+        const response = await fetch('https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery', {
+            method: 'POST',
+            headers: {
+                Accept : 'application/json; api-version=3.0-preview',
+                'Content-Type' : 'application/json'
+            },
+            body: '{"filters": [{"criteria": [{"filterType": 7, "value": "ms-vscode.cpptools"}]}], "flags": 529}'
+        });
+
+        const data = await response.json();
+        const preReleaseAvailable = data.results[0].extensions[0].versions[0].properties.some((e: object) => Object.values(e).includes("Microsoft.VisualStudio.Code.PreRelease"));
+
+        // If the user isn't on the pre-release version, but one is available, prompt them to install it.
+        if (preReleaseAvailable) {
+            displayedPreReleasePrompt.Value = true;
+            const message: string = localize("prerelease.message", "There is a new pre-release version of the C/C++ extension, would you like to install it?");
```

Comment:
This is a "comma splice" (two independent clauses).  Either:
a. Separate into two sentences.
b. Use a semicolon.
c. Use a conjunction. (less preferred).

My opinion would be to use 2 sentences.

Reply:
Updated the wording to use two sentences. Also open to completely rewording the prompt, I just slightly modified the old "switch to marketplace for pre-release" prompt to make sense in the general case, but if there's a more specific message we want to get a across I'm open to it.